### PR TITLE
Implement for Android

### PR DESCRIPTION
**Pull request by licy183**
Tested on x86_64-linux-android/aarch64-linux-android/arm-linux-androideabi.

### CODE REVIEW

File: src/tz_android.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// From https://android.googlesource.com/platform/ndk/+/android-4.2.2_r1.2/docs/system/libc/OVERVIEW.html
+// The name of the current timezone is taken from the TZ environment variable,
+// if defined. Otherwise, the system property named 'persist.sys.timezone' is
+// checked instead.
+
+const TIMEZONE_ENV_KEY: &str = "TZ";
+const TIMEZONE_PROP_KEY: &str = "persist.sys.timezone";
+
+pub(crate) fn get_timezone_inner() -> Result<String, crate::GetTimezoneError> {
+    return match env::var(TIMEZONE_ENV_KEY) {
```
Comment (Kijewski) [inline src/tz_android.rs:13]: Actually, we don't check the environment variables for any other OS, and unless Android is **very** special in this regard, I don't think we should do it in here, either. Cf: https://github.com/time-rs/time/security/advisories/GHSA-wcg3-cvx6-7396

File: src/tz_android.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        Ok(val) => Ok(val),
+        _ => {
+            let properties = AndroidSystemProperties::new();
+            if let Some(value) = properties.get(TIMEZONE_PROP_KEY) {
+                Ok(value)
+            } else {
+                Err(crate::GetTimezoneError::OsError)
+            }
```
Comment (Kijewski) [inline src/tz_android.rs:21]: This can be simplified to

```rust
properties.get(TIMEZONE_PROP_KEY).ok_or(crate::GetTimezoneError::OsError)
```

**Review by Kijewski — CHANGES_REQUESTED** [review]
Thank you for the PR! I only have two minor comments.

File: src/tz_android.rs (hunk 1 above)
Comment (licy183) [reply src/tz_android.rs:13]: I'm not sure about this. `TZ` environment is not set by default in Android, so let's assume it that `TZ` is always not set. I will remove this later.

File: src/tz_android.rs (hunk 2 above)
Comment (licy183) [reply src/tz_android.rs]: Ok, I'll fix this.

File: src/tz_android.rs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+use std::env;
+use android_system_properties::AndroidSystemProperties;
+
+// From https://android.googlesource.com/platform/ndk/+/android-4.2.2_r1.2/docs/system/libc/OVERVIEW.html
+// The name of the current timezone is taken from the TZ environment variable,
+// if defined. Otherwise, the system property named 'persist.sys.timezone' is
+// checked instead.
```
Comment (Kijewski) [inline src/tz_android.rs:7]: Could you please update the comment, too?

File: src/tz_android.rs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+const TIMEZONE_PROP_KEY: &str = "persist.sys.timezone";
+
+pub(crate) fn get_timezone_inner() -> Result<String, crate::GetTimezoneError> {
+    let properties = AndroidSystemProperties::new();
+    return properties.get(TIMEZONE_PROP_KEY).ok_or(crate::GetTimezoneError::OsError);
```
Comment (Kijewski) [inline src/tz_android.rs:13]: In Rust, the outcome of the last statement is also the result of the function. So if the last statement it `return something;`, you can simplify it to `something`:

```rust
pub(crate) fn get_timezone_inner() -> Result<String, crate::GetTimezoneError> {
    AndroidSystemProperties::new()
        .get(TIMEZONE_PROP_KEY)
        .ok_or(crate::GetTimezoneError::OsError)
}
```

**Review by Kijewski — CHANGES_REQUESTED** [review]

**Comment by astraw** [comment]
published in 0.1.40

**Comment by licy183** [comment]
Thanks!

**Pull request merged**